### PR TITLE
Hive: Remove deprecated AssertHelpers

### PR DESCRIPTION
**Pull request by liuxiaocs7**
- SubTask of https://github.com/apache/iceberg/issues/7094

Remove AssertHelpers in `iceberg-hive-metastore` module

### CODE REVIEW

File: hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveSchemaUtil.java [hunk 1]
```diff
@@ -147,13 +147,12 @@ public void testSchemaConvertToIcebergSchemaForEveryPrimitiveType() {
...
     for (FieldSchema notSupportedField : getNotSupportedFieldSchemas()) {
-      AssertHelpers.assertThrows(
-          "should throw exception",
-          IllegalArgumentException.class,
-          "Unsupported Hive type",
-          () -> {
-            HiveSchemaUtil.convert(Lists.newArrayList(Arrays.asList(notSupportedField)));
-          });
+      Assertions.assertThatThrownBy(
+              () -> {
```
Comment (nastra) [inline hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveSchemaUtil.java:151]: nit: can we remove the {} wrapping here?

File: hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java [hunk 2]
```diff
@@ -410,39 +403,40 @@ public void testCreateNamespaceWithOwnership() throws Exception {
...
-        });
+    assertThatThrownBy(
+            () -> {
+              try {
+                createNamespaceAndVerifyOwnership(
+                    "create_with_owner_type_alone",
+                    ImmutableMap.of(HiveCatalog.HMS_DB_OWNER_TYPE, PrincipalType.USER.name()),
+                    "no_post_create_expectation_due_to_exception_thrown",
+                    null);
+              } catch (TException e) {
```
Comment (nastra) [inline hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java:414]: I think we should remove those try-catch blocks as they are not needed

File: hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveSchemaUtil.java (hunk 1 above)
Comment (liuxiaocs7) [reply hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveSchemaUtil.java]: make sense, thanks!

File: hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java (hunk 2 above)
Comment (liuxiaocs7) [reply hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCatalog.java]: fixed, thanks!

**Review by nastra — APPROVED** [review]

**Pull request merged**